File: src/main/java/com/heekng/celloct/dto/StaffDto.java
package com.heekng.celloct.dto;

import com.heekng.celloct.entity.Staff;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;

public class StaffDto {

    @Getter
    public static class AddRequest {

        private Long shopId;
        private Long memberId;

        @Builder
        public AddRequest(Long shopId, Long memberId) {
            this.shopId = shopId;
            this.memberId = memberId;
        }
    }

    @Getter
    public static class UpdateEmploymentDateRequest {

        private Long staffId;
        private LocalDate changeEmploymentDate;

        @Builder
        public UpdateEmploymentDateRequest(Long staffId, LocalDate changeEmploymentDate) {
            this.staffId = staffId;
            this.changeEmploymentDate = changeEmploymentDate;
        }
    }

    @Getter
    @Setter
    public static class StaffResponse {
        private Long id;
        private String name;

        @Builder
        public StaffResponse(Staff staff) {
            this.id = staff.getId();
            this.name = staff.getName();
        }
    }

    @Getter
    @Setter
    public static class WithMemberResponse {
        private Long staffId;
        private String staffName;
        private Long memberId;
        private String memberName;
        private String email;

        @Builder
        public WithMemberResponse(Staff staff) {
            this.staffId = staff.getId();
            this.staffName = staff.getName();
            this.memberId = staff.getMember().getId();
            this.memberName = staff.getMember().getName();
            this.email = staff.getMember().getEmail();
        }
    }

    @Getter
    @Setter
    public static class DeleteRequest {
        private Long staffId;
        private Long shopId;
        private Long memberId;

        @Builder
        public DeleteRequest(Long staffId, Long shopId, Long memberId) {
            this.staffId = staffId;
            this.shopId = shopId;
            this.memberId = memberId;
        }
    }

    @Getter
    @Setter
    public static class UpdateRequest {
        private Long staffId;
        private String name;
        private Long shopId;
        private Long memberId;
        private String staffEmploymentDate;
    }

    @Getter
    @Setter
    public static class StaffDetailResponse {
        private Long id;
        private String name;
        private LocalDate employmentDate;

        @Builder
        public StaffDetailResponse(Staff staff) {
            this.id = staff.getId();
            this.name = staff.getName();
            this.employmentDate = staff.getEmploymentDate();
        }
    }


}

File: src/main/kotlin/com/heekng/celloct/dto/StaffDto.kt
package com.heekng.celloct.dto

import com.heekng.celloct.entity.Staff
import java.time.LocalDate

class StaffDto {
    data class AddRequest (val shopId: Long, val memberId: Long)

    data class UpdateEmploymentDateRequest (
        val staffId: Long,
        val changeEmploymentDate: LocalDate
    )

    data class StaffResponse (
        val id: Long?,
        val name: String,
    ) {
        constructor(staff: Staff): this(
            id = staff.id,
            name = staff.name,
        )
    }

    data class WithMemberResponse (
        val staffId: Long?,
        val staffName: String,
        val memberId: Long?,
        val memberName: String,
        val email: String,
    ) {
        constructor(staff: Staff): this(
            staffId = staff.id,
            staffName = staff.name,
            memberId = staff.member.id,
            memberName = staff.member.name,
            email = staff.member.email,
        )
    }

    data class DeleteRequest (
        val staffId: Long,
        val shopId: Long,
        val memberId: Long
    )

    data class UpdateRequest (
        var staffId: Long,
        val name: String,
        var shopId: Long,
        val memberId: Long? = null,
        val staffEmploymentDate: String? = null,
    )

    data class StaffDetailResponse (
        val id: Long?,
        val name: String,
        val employmentDate: LocalDate,
    ) {
        constructor(staff: Staff): this(
            id = staff.id,
            name = staff.name,
            employmentDate = staff.employmentDate,
        )
    }
}